src/AudioGeneratorMOD.cpp

```diff
@@ -800,27 +806,50 @@ void AudioGeneratorMOD::GetSample(int16_t sample[2])
 
     out = current;
 
-    // Integer linear interpolation
+    // Integer linear interpolation - only works correctly in 16bit
     out += (next - current) * (Mixer.channelSampleOffset[channel] & ((1 << FIXED_DIVIDER) - 1)) >> FIXED_DIVIDER;
 
     // Upscale to BITDEPTH
-    out <<= BITDEPTH - 8;
+    out32 = (int32_t)out << (BITDEPTH - 8);
 
     // Channel volume
-    out = out * Mixer.channelVolume[channel] >> 6;
+    out32 = out32 * Mixer.channelVolume[channel] >> 6;
 
     // Channel panning
-    sumL += out * min(128 - Mixer.channelPanning[channel], 64) >> 6;
-    sumR += out * min(Mixer.channelPanning[channel], 64) >> 6;
+    sumL += out32 * min(128 - Mixer.channelPanning[channel], 64) >> 6;
+    sumR += out32 * min(Mixer.channelPanning[channel], 64) >> 6;
   }
 
-  // Downscale to BITDEPTH
-  sumL /= Mod.numberOfChannels;
-  sumR /= Mod.numberOfChannels;
+  // Downscale to BITDEPTH - original code
+  //sumL /= Mod.numberOfChannels;
```

Comment:
Let's just throw out the old code, please.  GIT history will have it, no reason for it to live in comments in the fixed version...

Reply:
Thanks for the review - I'll update my PR as you suggest.